test-harness/src/tracing_reader.rs

```diff
@@ -0,0 +1,76 @@
+use std::path::Path;
+use std::time::Duration;
+
+use backoff::backoff::Backoff;
+use backoff::ExponentialBackoff;
+use miette::Context;
+use miette::IntoDiagnostic;
+use tokio::fs::File;
+use tokio::io::AsyncBufReadExt;
+use tokio::io::BufReader;
+use tokio::io::Lines;
+use tokio::sync::mpsc;
+use tracing::instrument;
+
+use super::Event;
+
+/// A task to read JSON tracing log events output by `ghid-ng` and send them over a channel.
+#[allow(dead_code)]
+pub struct TracingReader {
+    sender: mpsc::Sender<Event>,
+    lines: Lines<BufReader<File>>,
+}
+
+impl TracingReader {
+    /// Create a new [`TracingReader`], sending log events to the given `sender` after they're
+    /// written to the given `path`.
```

Comment:
the description sounds like the file is meant to be written to, but the file is wrapped in a `BufReader`

Reply:
Clarified this, PTAL